arch/riscv/core/stacktrace.c

```diff
@@ -88,15 +88,20 @@ static inline bool in_text_region(uintptr_t addr)
 #ifdef CONFIG_FRAME_POINTER
 void z_riscv_unwind_stack(const struct arch_esf *esf)
 {
-	uintptr_t fp = esf->s0;
+	uintptr_t fp, last_fp = 0;
 	uintptr_t ra;
 	struct stackframe *frame;
 
+	if (esf != NULL) {
```

Comment:
why did you move this check to the `unwind` function? Before, this was checked before calling the function, now we have to have this check in two places - as there are two implementations of `z_riscv_unwind_stack`